crates/sui-indexer-alt-jsonrpc/src/api/transactions/mod.rs

```diff
@@ -0,0 +1,159 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use futures::future;
+use jsonrpsee::{core::RpcResult, proc_macros::rpc};
+use serde::{Deserialize, Serialize};
+use sui_json_rpc_types::{
+    Page, SuiTransactionBlockResponse, SuiTransactionBlockResponseOptions,
+    SuiTransactionBlockResponseQuery,
+};
+use sui_open_rpc::Module;
+use sui_open_rpc_macros::open_rpc;
+use sui_types::digests::TransactionDigest;
+
+use crate::{context::Context, error::InternalContext};
+
+use super::rpc_module::RpcModule;
+
+mod error;
+mod filter;
+mod response;
+
+#[open_rpc(namespace = "sui", tag = "Transactions API")]
+#[rpc(server, namespace = "sui")]
+trait TransactionsApi {
+    /// Fetch a transaction by its transaction digest.
+    #[method(name = "getTransactionBlock")]
+    async fn get_transaction_block(
+        &self,
+        /// The digest of the queried transaction.
+        digest: TransactionDigest,
+        /// Options controlling the output format.
+        options: SuiTransactionBlockResponseOptions,
+    ) -> RpcResult<SuiTransactionBlockResponse>;
+}
+
+#[open_rpc(namespace = "suix", tag = "Query Transactions API")]
+#[rpc(server, namespace = "suix")]
+trait QueryTransactionsApi {
```

Comment:
Why having a separate trait for `queryTransactionBlocks` instead of having it as part of `TransactionsApi`?

Reply:
It's unfortunate -- it's because of the namespace, which is `suix` for this, and `sui` for the `getTransactionBlock`.